front/components/app/DatasetView.tsx

```diff
@@ -423,171 +425,203 @@ export default function DatasetView({
     datasetTypes,
   ]);
 
+  useEffect(() => {
```

Comment:
That's keeping the state up-to-date after a dataset switch - I think it could be avoided using SWR but in all of dust apps it looks like we're not using SWR hooks, so I kept the same style (I believe?)

Reply:
Yep but it blocks the main rendering loop here the block appearance can become extremely slow

Agreed there is no SWR in Dust apps because it's the oldest part of the code but I thiiink this is what we want as we update this behavior?

Reply:
Ok ! I'll need to dive in more into SWR then, will look 👀